File: Guess_Random_Number.py
# Guessing Game
# File name: Guess_Random_Number.py
# Author: Natchathiram Premkumar
# Date: 16/07/2021, Initial version

from random import randrange, uniform

# randrange gives you an integral value
RandomNo = randrange(1, 100)

# Initializing the number of guesses.
count = 0
MaxCount = 10
Left = MaxCount
while count < MaxCount:
	count += 1
	Left  -= 1

	# taking guessing number as input
	UserInput = int(input("Enter guess what the target number is:- "))

	# Condition testing
	if RandomNo == UserInput:
		print("Good Job! You guessed it!")
		
		# Once guessed, loop will break
		break
		
	elif RandomNo > UserInput:
		print("Oops, Your guess was LOW.")
		print("You have only left",Left,"Chances")
		
	elif RandomNo < UserInput:
		print("Oops, Your guess was HIGH.")
		print("You have only left",Left,"Chances")

if count >= MaxCount:
	print("\nSorry, You didn't guess my number. It was %d" % RandomNo)

File: Guess_Random_Number.go
// Guessing Game
// File name: Guess_Random_Number.go
// Author: Natchathiram Premkumar
// Date: 16/07/2021, Initial version

package main
 
import(
   "fmt"
   "math/rand"
   "time"
)
 
func xrand(min, max int) int {
   rand.Seed(time.Now().Unix())
   return rand.Intn(max - min) + min
}
 
func main() {
   mynumber := xrand(1, 100)
   fmt.Println("You guessed the number in %d tries\n", mynumber)
   attempts := 0
   left := 10
   var guess int
   fmt.Println("Guess a number between 1 and 100.")
 
   for attempts < 10 {
	  attempts++
	  left--
      fmt.Println("Enter your guess of what the target number is:")
      fmt.Scanf("%d", &guess)

      if guess < mynumber {
         fmt.Println("Oops, Your guess was LOW.")
		 fmt.Println("You have only left", left, "Chances")
      } else if guess > mynumber {
         fmt.Println("Oops, Your guess was HIGH.")
		 fmt.Println("You have only left", left, "Chances")
      } else if guess == mynumber {
         break
      }
   }
   if guess == mynumber {
      fmt.Printf("Good Job! You guessed it!")
   } else {
      fmt.Printf("Sorry, You didn't guess my number. It was %d", mynumber)
   }
}